src/components/AddAssignment.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { Link } from 'react-router-dom'
+import Cookies from 'js-cookie';
+import { Button, } from '@mui/material';
+import Radio from '@mui/material/Radio';
+import { DataGrid } from '@mui/x-data-grid';
+import { SERVER_URL } from '../constants.js'
+
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+
+const AddAssignment = () => {
+
+    const [name, setname] = React.useState("")
+    const [date, setDate] = React.useState("")
+    const [course, setCourse] = React.useState("")
+
+    const handleChangeName = (e) => {
+        console.log("Hello", e.target.value)
+        setname(e.target.value)
+
+    }
+
+    const handleChangeDate = (e) => {
+        console.log("Date", e.target.value)
+
+        setDate(e.target.value)
+    }
+
+    const handleChangeCourse = (e) => {
+        console.log("course", e.target.value)
+
+        setCourse(e.target.value)
+    }
+
+    const fetchAssignments = () => {
+        console.log("Assignment.fetchAssignments");
+        const token = Cookies.get('XSRF-TOKEN');
+        fetch(`${SERVER_URL}/gradebook`,
+            {
+                method: 'GET',
+                headers: { 'X-XSRF-TOKEN': token }
+            })
+            .then((response) => response.json())
+            .then((responseData) => {
+                if (Array.isArray(responseData.assignments)) {
+                    //  add to each assignment an "id"  This is required by DataGrid  "id" is the row index in the data grid table 
+                    this.setState({ assignments: responseData.assignments.map((assignment, index) => ({ id: index, ...assignment })) });
+                } else {
+                    toast.error("Fetch failed.", {
+                        position: toast.POSITION.BOTTOM_LEFT
+                    });
+                }
+            })
+            .catch(err => console.error(err));
+    }
+
+    const addAssignment = () => {
+        console.log("add assignment")
+        const params = {
```

Comment:
nit: may be  call it `body` or `bodyObject` since params usually refer to query parameters - i.e. GET request.

Reply:
got it, will add it to the list of changes